### DISCUSSION TITLE
Open dds security

### DISCUSSION THREAD

**Discussion by gajananmhaske** [OpenDDS/OpenDDS#3605] [Q&A] [answered]
/opendds3.20/tests/security/attributes/./subscriber 
infodiscovery error am getting same error getting for publisher.

/opendds3.20/tests/security/attributes/./run_test.pl

publisher exiting  -11
subscriber exiting -21.

both are create participants error.

**Comment by mitza-oci** [comment]
Please see the OpenDDS developer's guide section "14.5 Required OpenDDS Configuration" (for DDS Security). InfoRepo Discovery is not used in security-enabled applications.
If you're running one of the included tests please attach a plain text file with the output.

**Reply by gajananmhaske** [reply]
Thank adam for revert back i will check on that.

**Reply by gajananmhaske** [reply]
hello adam,

i have fixed that info repo error.

Now i am getting issue:

if(Theserviceparticipant->get_security())
this is not woking.

OPENDDS_SECURITY   --  This micro also not setting.
can you please us how to set this or where we will get  information about this.

Opendds3.20/tests/security/attributes/publisher.cpp

above path i am executing my code.

**Reply by mitza-oci** [reply]
Please run `tools/scripts/show_build_config.pl` and post the output so we can see how OpenDDS is configured

**Reply by gajananmhaske** [reply]
```



......
...........................setenv.h.........................................

./configure --security --xerces --openssl


...............usr_micro.GNU.........................
ssl=1
openssl11=1
no_opendds_security=0
..........------------------------------------------------------

.........................VERSION.txt.............--------------
opendds 3.20 released on march 25 2022

#define ACE_VERSION   6.2a_p24
........................default.feature.........
no_cxx11=0
ssl=1
xerces3=1
openssl11=1
no_opendds_security=0

........platform macro.GNU...........
optimize=0
xerces3=1
include $(ACEROOT)/include/makeinclude/platform_linux.GNU

................config.h.......\

#include ace/config-linux.h
```

**Reply by mitza-oci** [reply]
1. Remove all generated files or start from a new working directory and get OpenDDS source code from git or by expanding a release archive
2. Configure with `--tests` in addition to the options you've used before
3. When you build the workspace the code in `tests/security/attributes` will be included.

**Comment by Ayan114189** [comment]
Hi @mitza-oci,
In our sec_base.ini file "DPSSecurity=1" is mentioned.
DiscoveryConfig=uni_rtps
transport_type=rtps_udp

Still it is not working.

**Reply by jrw972** [reply]
Please post the exact command lines you are using and a copy of the logs.

**Reply by Ayan114189** [reply]
We are running ./run_test.pl "SC0_sec_on" from OpenDDS-3.20/tests/security/attributes/
And getting the error mentioned above.

I am unable to attach any document in this site, there is no option to insert. Drag and drop is also not working. Basically wanted to share the publisher.log and subscriber.log files

**Reply by jrw972** [reply]
Try creating a gist https://gist.github.com/

**Comment by gajananmhaske** [comment]
./Configure  --tests --security --xerces --openssl

Error : Google test 'googletest/include/gtest' not found in submodule src
'Opendds/tests/googletest/Googletest'
Or default install dir ''/usr" please pass the correct version of Google
Test to --gtest

**Reply by mitza-oci** [reply]
Some tests use the GoogleTest libraries. See https://github.com/objectcomputing/OpenDDS/blob/master/docs/dependencies.md#google-test
If you're using git it's available as a submodule.

**Reply by gajananmhaske** [reply]
Thanks for help and support it was good learning for me.

I am trying to configure opendds
./configure --prefix=PATH --test --security --xerces --openssl

after make we are trying make install

its not working giving error on line no GNUmake.file.ACE:531 : receip for target install failed.

ERROR: Enable to copy README file to folder PATH we are slected.

not able to create that lib folder.

please help us regarding same.

**Reply by jrw972** [reply]
PATH needs to be valid file system path, i.e., the place where you would like to install, e.g., /usr/local

**Reply by gajananmhaske** [reply]
yes, PATH i have given 
home/me/opendds_build

**Comment by gajananmhaske** [comment]
Thanks for your guidance.
 ./configure --tests --security --xerces --openssl  It works fine
 I have checked with ./run_test.pl "SC0_sec_on"  It's works.
But when i have signed that's certs folder using openssl coomand given in
README file  Then test failed.......
And

  Also one more query how we know that that communication taking place
using security?
  PFA test failed after signed documents.

**Reply by jrw972** [reply]
The security tests use DDS Security documents that have be prepared to be consistent.
Participant certificates must be signed by the Identity CA.
The governance file and permissions file must be signed by the Permissions CA.
If you are resigning these to learn about security, then you must use the correct CA.

The security attributes tests are design to expect a certain number of messages.
They will fail if these messages are not received.

**Comment by gajananmhaske** [comment]
while running binary file using ./filename -DCPSConfigFile --sec_base.ini &

when running _ Inforepo_discovary error getting please
 help us

**Reply by jrw972** [reply]
Security is only support with RTPS discovery and transport.

**Comment by gajananmhaske** [comment, marked as answer]
Thank you for the clarification.
